maths/binary_exponent.py

```diff
@@ -11,24 +12,28 @@ def bin_exp(num: int, expo: int) -> int:
     1
     >>> bin_exp(-2, -2)
     0
-    >>> bin_expo(1.5, 2.5)
+    >>> bin_exp(1.5, 2.5)
     2
-    >>> bin_expo('a', 'b')
-    TypeError: can't multiply sequence by non-int of type 'str'
-    >>> bin_expo(5, None)
-    TypeError: can't multiply sequence by non-int of type 'str'
-  
+    >>> bin_exp('a', 'b')
+    'An exception occurred'
+    >>> bin_exp(5, None)
+    1
     """
+    
     result = 1
     while expo:
-        div, mod = divmod(expo, 2)
-        if mod:
-            result *= num
-            expo -= 1
-        else:
-            num *= num
-            expo = div
-    return result
+        try:
+            div, mod = divmod(expo, 2)
+            if mod:
+                result *= num
+                expo -= 1
+            else:
+                num *= num
+                expo = div 
+            return result  
+        except:
```

Comment:
Bare exceptions are not accepted in this repo.  https://realpython.com/the-most-diabolical-python-antipattern